### PR TITLE
[IOTDB-2276] Optimize aligned query with value filter to accelerate it and reduce its memory footprint

### PR DESCRIPTION
**Pull request by JackieTien97**
Previously, raw query with value filter won't use cache in where clause, in this pr, we use it.
Aggregate query with value filter has something wrong with the `MetaUtils.groupAlignedSeriesWithAggregations`, it will change the `selectedSeries` which will casue it won't use cache either

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/path/AlignedPath.java [hunk 1]
```diff
@@ -201,6 +218,7 @@ public PartialPath copy() {
     result.fullPath = fullPath;
     result.device = device;
     result.measurementList = new ArrayList<>(measurementList);
```
Comment (Alima777) [inline server/src/main/java/org/apache/iotdb/db/metadata/path/AlignedPath.java:220]: Add ` result.isUnderAlignedEntity = isUnderAlignedEntity;`

Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/metadata/path/AlignedPath.java:220]: MeasurementPath's copy() method need to be modified too.

Comment (JackieTien97) [reply server/src/main/java/org/apache/iotdb/db/metadata/path/AlignedPath.java:220]: `isUnderAlignedEntity` is a filed of `MeasurementPath`, not `AlignedPath`.

Comment (JackieTien97) [reply server/src/main/java/org/apache/iotdb/db/metadata/path/AlignedPath.java:220]: > MeasurementPath's copy() method need to be modified too.

done.

**Review by Alima777 — APPROVED** [review]
LGTM

**Pull request merged**